File: Napakalaki Ruby/CultistPlayer.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  

require_relative 'Monster.rb'
require_relative 'Player.rb'

class CultistPlayer < Player
  
  @@totalCultistPlayers =0
  
  def initialize(p_aux,c)
    
    copiarJugador(p_aux)
    @myCultistCard = c
    @@totalCultistPlayers = @@totalCultistPlayers +1
    puts 'Te has convertido en cultist'
  end
  
  protected
  
  def getOponentLevel(m)
    n = m.getCombatLevelAgainstCultistPlayer
    return n
  end
  
  def shouldConvert()
    aux = false
    return aux
  end
  
  def getCombatLevel()
    aux = super
    n = aux + aux*0.2 + (@myCultistCard.getGainedLevels)*@@totalCultistPlayers
    n= n/1
    
    return n
  end
  
  public
  
  def giveMeATreasure()
    tesoro = @visible_treasures[rand(@visible_treasures.size)]
    @visible_treasures.delete_at(@visible_treasures.index(tesoro))
    return tesoro
  end
  
  def canYouGiveMeATreasure()
    can_give = false
    can_give = true unless (@visible_treasures.length < 1)
    return can_give                        
  end
  
  
  def self.getTotalCultistPlayers
    return @@totalCultistPlayers
  end
    
end

end

File: Napakalaki Java/NapakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author Germán
 */
public class CultistPlayer extends Player {
    private static int totalCultistPlayers = 0;
    private Cultist myCultistCard = null;
    
    public CultistPlayer(Player p, Cultist c){
        super(p);
        //System.out.println("Se convierte en sectario\n");
        totalCultistPlayers++;
        myCultistCard = c;
    }
   
    @Override
    protected boolean shouldConvert() {
        return false;
    }
    
    @Override
    public int getCombatLevel(){
        double nivel;
        nivel = super.getCombatLevel() + super.getCombatLevel()*0.2;
        nivel += myCultistCard.getGainedLevels()*totalCultistPlayers;
        return (int) nivel;
    }
    
    @Override
    protected int getOponentLevel(Monster m) {
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    public static int getTotalCultistPlayers() {
        return totalCultistPlayers;
    }
    
    @Override
    protected Treasure giveMeATreasure() {
        Treasure tesoro;
        ArrayList<Treasure> tesoros = getVisibleTreasures();
        tesoro = tesoros.get((int) (Math.random()*tesoros.size()));
        tesoros.remove(tesoro);
        super.setVisibleTreasures(tesoros);
        return tesoro;
    }
    
   @Override 
   public boolean canYouGiveMeATreasure() {
       return !(getVisibleTreasures().isEmpty());
   }
            
    
}
